Write QR code cache to JSON file on shutdown

brendan-ly opened the pull request:
* Added --cache-state-file argument 
* Uses JSON file to store qr codes cache
* Server shutdown now writes to JSON file instead of deleting cache
* Server now reads cache from JSON file on startup

brendan-ly committed:
wrote qr code cache state to JSON file on server
shutdown

evanugarte commented on modules/args.py:
```diff
@@ -59,5 +59,9 @@ def get_args():
         help="the base url path for the center image in qr code"
     )
-    
+    parser.add_argument(
+        "--cache-state-file",
+        required=True,
+        help="the JSON file where the cache will map to"
+    )
```
can we make this optional? if the user doesnt specify it, we just clear the cache

evanugarte commented on modules/qr_code.py:
```diff
@@ -87,3 +92,25 @@ def clear(self):
+        except FileNotFoundError:
+            logger.exception(f"Could not find cache state file: {self.cache_state_file}")
+        except Exception:
+            logger.exception(f"An unexpected error occurred while saving cache state file: {self.cache_state_file}")
```
add a blank newline to the bottom of this file, so the red icon goes away

brendan-ly committed:
cache-state-file arg is optional
if the user doesnt specify it, cache is cleared
if the user specifies it, cache is saved to that file

brendan-ly committed:
default val of cache_state_file parameter is now None

evanugarte commented on modules/args.py:
```diff
@@ -59,5 +59,9 @@ def get_args():
-    
+    parser.add_argument(
+        "--qr-code-cache-state-file",
+        default=None,
```
```suggestion
```

evanugarte commented on modules/qr_code.py:
```diff
@@ -87,3 +93,25 @@ def clear(self):
+        try:
+            with open(self.cache_state_file, 'w') as json_file:
+                json.dump(self.mapping, json_file)
+        except FileNotFoundError:
+            logger.exception(f"Could not find cache state file: {self.cache_state_file}")
```
i think we can remove this

evanugarte commented on server.py:
```diff
@@ -178,9 +179,13 @@ def get_metrics():
 @app.on_event("shutdown")
 def signal_handler():
-    qr_code_cache.clear()
+    if args.qr_code_cache_state_file is None:
+        qr_code_cache.clear()
+    else:
+        qr_code_cache.write_cache_state()
```
```suggestion
    if args.qr_code_cache_state_file is None:
        return qr_code_cache.clear()
    
    qr_code_cache.write_cache_state()
```

brendan-ly committed:
update cache file arg and shutdown logic

evanugarte committed:
add --qr-code-cache-state-file=/tmp/cleezy-state

evanugarte committed:
lint city

evanugarte committed:
idk

evanugarte committed:
remove debug log

brendan-ly committed:
fixed broken sce_logo.png

cuzethan reviewed (approved):
looks good
